File: Backend/Contracts/Contracts.cs
namespace Backend.Contracts;

public class Conversation
{
  public required string Title { get; set; }
  public required List<ChatMessage> Messages { get; set; }
  public required string Guid { get; set; }
  public required bool IsFollowed { get; set; } = false;
  public required string UserId { get; set; } = string.Empty;
}

public record ChatMessage(ChatRole Role, string Content, string? Name = "");

public record UserInformation(string Id, string DisplayName);

public enum ChatRole
{
  Assistant,
  User,
  System,
  Function,
  Tool
}

File: src/lib/models/Contracts.ts
export interface Conversation {
  id: string;
  title: string;
  messages: ChatMessage[];
  isFollowed: boolean;
  userId: string;
  date: Date;
  citations: Citation[];
  useDocumentSearch: boolean;
}

export interface ChatMessage {
  role: ChatRole;
  content: string;
  name: string;
  context: { messages: ChatMessage[] } | undefined;
}

export interface ToolMessage {
  citations: Citation[];
}

export interface Citation {
  id: number;
  title: string;
  url: string;
  content: string;
}

export interface UserInformation {
  id: string;
  displayName: string;
}

export enum ChatRole {
  Assistant = 'assistant',
  User = 'user',
  System = 'system',
  Function = 'function',
  Tool = 'tool'
}
export interface State {
  useHistory: boolean;
  useMock: boolean;
  autosave: boolean;
  version: string;
  sidebarSlot: string;
  deployment?: string;
  chatMode?: ChatMode;
}

export interface ClientPrincipal {
  auth_typ: string;
  claims: { typ: string; val: string }[];
  name_typ: string;
  role_typ: string;
}

export enum ChatMode {
  Balanced = 'balanced',
  Creative = 'creative',
  Precise = 'precise'
}

export interface SelectOption {
  label: string;
  value: string;
}
